code/game/machinery/shieldgen.dm

```diff
@@ -173,7 +178,7 @@
 
 /obj/machinery/shieldgen/proc/checkhp()
 	if(health <= 30)
-		src.malfunction = 1
+		malfunction = 1

```

Comment:
Assuming malfunction is only a 1 or 0 value var, you could swap this to TRUE / FALSE for readability